src/main/java/org/traccar/protocol/TranSyncProtocolDecoder.java

```diff
@@ -0,0 +1,353 @@
+package org.traccar.protocol;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.channel.Channel;
+import org.traccar.BaseProtocolDecoder;
+import org.traccar.Protocol;
+import org.traccar.helper.BitUtil;
+import org.traccar.helper.UnitsConverter;
+import org.traccar.model.CellTower;
+import org.traccar.model.Network;
+import org.traccar.model.Position;
+import org.traccar.session.DeviceSession;
+
+import java.net.SocketAddress;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+
+public class TranSyncProtocolDecoder extends BaseProtocolDecoder {
+
+    public TranSyncProtocolDecoder(Protocol protocol) {
+        super(protocol);
+    }
+
+    private boolean isOptionalParameters;
+    private boolean extended;
+    private final String STX = "3A3A";
+    private final String ETX = "2323";
+    private String lac;
+    private String deviceId;
+    private Date datePacket;
+    private double latitude;
+    private double longitude;
+    private double speed;
+    private int course;
+    private int mobileNetworkCode;
+    private int gsmSignalStrength;
+    private double batteryVoltage;
+    private int satellitesNumber;
+    private int hdopProtocol;
+    private short adcVoltageInMilliVolts;
+    private String rfidTagName;
+    private int adc2;
+    private int cellIdInt;
+    private int lacInt;
+    private int odometer;
+    private boolean isGpsFix;
+    private boolean isLiveData;
+    private boolean hasGpsAlert;
+    private boolean isIgnitionOff;
+    private boolean isPowerOff;
+    private int deviceAlert;
+    private String gpsTrackerModel;
+    private boolean isOutPutTwo;
+    private boolean isOutPutOne;
+    private boolean isInPutThree;
```

Comment:
Why do we need all these instance variables?

Reply:
I`m not really sure about instance variables. Just translate it from Python class.